### ISSUE TITLE
Implement boolean operations into the library

### ISSUE THREAD

**Issue by emil-peters** [mikedh/trimesh#1947]
When shipping code I can't force people to have blender or SCAD installed onto their machine.
Would it be possible to have boolean operation build into the library instead of having them using a different engine?

Kind regards

**Comment by mikedh** [comment]
Hey, yeah I would also like to drop or reduce the scope of `trimesh.interfaces` to OpenSCAD/blender if at all possible. Until recently the state of boolean operation backends available has been quite bad (unreliable and slow). I haven't tested but this looks very promising and the OpenSCAD merge is a pretty good sign: https://github.com/elalish/manifold/issues/361

I think the road to getting that as an `trimesh[easy]` dependency is to [set up cibuildwheel](https://github.com/mikedh/embreeX/blob/main/pyproject.toml) and to release a matrix [of Python 3.6-3.latest x Mac/Win/Linux](https://github.com/mikedh/embreeX/releases/tag/2.17.7.post1) and I'm sure they'd love packaging help 😄.

**Comment by pca006132** [comment]
@mikedh yes we would love some python packaging help, we are unable to get the windows build working: https://github.com/elalish/manifold/issues/362
